### PR TITLE
feat: new split_to_map udf

### PR DESCRIPTION
**Pull request by blueedgenick**
### Description 
New UDF  `split_to_map(input, entryDelimiter, kvDelimiter)` to build a map from a string. 

Useful for taking messages from upstream systems and converting them into a more structured and usable format.

### Testing done 
New Unit & QTT tests.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitToMap.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    if (input == null || entryDelimiter == null || kvDelimiter == null) {
+      return null;
+    }
+
+    if (entryDelimiter.isEmpty() || kvDelimiter.isEmpty() || entryDelimiter.equals(kvDelimiter)) {
+      return null;
+    }
+
+    final Iterable<String> entries = Splitter.on(entryDelimiter).omitEmptyStrings().split(input);
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitToMap.java:50]: by the way, guava `Splitter` has support for `withKeyValueSeparator` to simplify exactly this :)

From the docs:

```java
    /*
     * For example, {@code
     * Splitter.on(';').trimResults().withKeyValueSeparator("=>").split("a=>b ; c=>b")} will return
     * a mapping from {@code "a"} to {@code "b"} and {@code "c"} to {@code "b"}.
```

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitToMap.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.stream.StreamSupport;
+
+@UdfDescription(
+    name = "split_to_map",
+    description = "Splits a string into key-value pairs and creates a map from them. The "
+        + "'entryDelimiter' splits the string into key-value pairs which are then split by "
+        + "'kvDelimiter'. If the same key is present multiple times in the input, the latest "
+        + "value for that key is returned. Returns NULL f the input text or either of the "
+        + "delimiters is NULL.")
+public class SplitToMap {
```
Comment (agavra) [inline ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitToMap.java:32]: We may also want to have `EncodeMap(map, entry_delim, kv_delim)` which encodes a map into a string

File: ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitToMap.java (hunk 1 above)
Comment (blueedgenick) [reply ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/string/SplitToMap.java:50]: yeah, tried multiple ways with that at first but it's too inflexible and likes to throw a lot of exceptions on my test-cases :(

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 3]
```diff
@@ -701,6 +701,23 @@ If the delimiter is found at the beginning or end
...
 then an empty space is added to the array.
 
+### `SPLIT_TO_MAP`
+
+```sql
+SPLIT_TO_MAP(input, entryDelimiter, kvDelimiter)
+```
+
+Splits a string into key-value pairs and creates a map from them. The 
+'entryDelimiter' splits the string into key-value pairs which are then split by 'kvDelimiter'. If the same key is present multiple times in the input, the latest value for that key is returned.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:711]:
```suggestion
`entryDelimiter` splits the string into key-value pairs which are then split by `kvDelimiter`. If the same key is present multiple times in the input, the latest value for that key is returned. 
```

File: docs/developer-guide/ksqldb-reference/scalar-functions.md [hunk 4]
```diff
@@ -701,6 +701,23 @@ If the delimiter is found at the beginning or end
...
+### `SPLIT_TO_MAP`
+
+```sql
+SPLIT_TO_MAP(input, entryDelimiter, kvDelimiter)
+```
+
+Splits a string into key-value pairs and creates a map from them. The 
+'entryDelimiter' splits the string into key-value pairs which are then split by 'kvDelimiter'. If the same key is present multiple times in the input, the latest value for that key is returned. 
+
+Returns NULL f the input text is NULL.
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/scalar-functions.md:713]:
```suggestion
Returns NULL if the input text is NULL.
```

File: ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/SplitToMapTest.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.util.Collections;
+import java.util.Map;
+import org.junit.Test;
+
+public class SplitToMapTest {
```
[the remark this answers is not in the document]
Comment (blueedgenick) [reply ksqldb-engine/src/test/java/io/confluent/ksql/function/udf/string/SplitToMapTest.java:28]: good idea, test added!

**Pull request merged**